var/spack/repos/builtin/packages/fmt/package.py

```diff
@@ -63,11 +63,8 @@ def cmake_args(self):
         spec = self.spec
         args = []
 
-        if '+pic' in spec:
-            args.extend([
-                '-DCMAKE_C_FLAGS={0}'.format(self.compiler.cc_pic_flag),
-                '-DCMAKE_CXX_FLAGS={0}'.format(self.compiler.cxx_pic_flag)
-            ])
+        if self.spec.satisfies('+shared'):
+            args.append('BUILD_SHARED_LIBS=ON')
```

Comment:
I just gave this a try and it should be `-DBUILD_SHARED_LIBS=TRUE` (missing `-D`, not sure about `ON` vs. `TRUE`), otherwise the static library will be built.

Reply:
Thanks for giving this a test, indeed. `cmake` treats `ON`, `On`, `TRUE`, `True`, ``YES`, `Yes` all the same, it's a logical `1`, but the `-D` is indeed missing.

As this shows that this wasn't even tested or used by the submitter, I'd ask him/her to justify why he submitted adding the shared variant at all.

Reply:
CMS experiment is planning to use Spack instead of their own (RPM-based) system, and I'm working on porting their recipes to Spack where possible (for some packages, e.g. ROOT or DD4Hep, they have their own forks with their own patches that were rejected upstream). Yes, sorry for missing the `-D` part. And thanks for clarifying the `-fPIC` part. I will update the PR now following your recomendations.

Do you want to build both static and dynamic libraries **at the same time**, or just have a switch to choose one or another?

Reply:
The former is a bit more difficult to implement

Reply:
> Do you want to build both static and dynamic libraries **at the same time**, or just have a switch to choose one or another?

Thinking of it, spack can have both a `static` and a `shared` variant of the same library installed at the same time, spack can do both by installing both variants at the same time.

I've reviewed the cumulated change with your new commit ee26f81, and it looks good. I could approve that. I'll leave the rest for @michaelkuhn

Reply:
> spack can do both by installing both variants at the same time.

It depends on the package: sometimes `static` and `shared` does not override each other. For `fmt` it is not the case, however one can write a `@run_after('install')` hook and redo the build with different value of `BUILD_SHARED`, but this will probably require patching `fmt-targets.cmake`.